### DISCUSSION TITLE
can `set completeopt=menu,menuone,noselect,noinsert` work the same in vscode with the extension as it does in neovim?

### DISCUSSION THREAD

**Discussion by jtmkrueger** [vscode-neovim/vscode-neovim#1544] [Q&A] [answered]
I'd like tab completion to work just like it does in (neo)vim when I have these settings. 
My line to set it the way I like in my configs is
```
set completeopt=menu,menuone,noselect,noinsert
```
It's nice to just cycle through pressing tab until the correct choice is there and not have to ever hit enter. 

I haven't figured out a way to keep this consistent, but maybe there's something obvious I'm missing in the documentation or someone has already concocted an incantation to remedy this.

**Comment by xiyaowong** [comment]
No, it's impossible to support these options. You should use VSCode keybindings.

For example, accept the selected suggestion using tab

```json
  {
    "key": "tab",
    "command": "acceptSelectedSuggestion",
    "when": "suggestWidgetVisible && textInputFocus"
  }
```

In most cases, you should first check if VSCode can achieve the functionality you want, and then consider the features of nvim.

**Comment by jtmkrueger** [comment, marked as answer]
ah, okay. 
> You should use VSCode keybindings.

I already tried setting things up and didn't like how it just accepts on tab instead of letting you cycle through with tab and changing the text it put in as you cycle. It's not how vim works so wanted parity.
Sounds like maybe it just can't do what I want.

**Comment by theol0403** [comment]
https://github.com/microsoft/vscode/issues/70532

**Reply by jtmkrueger** [reply]
Thanks for pointing me at this, I hope it gets implemented natively

**Comment by gaoqiangks** [comment]
According to my understanding, almost all insert mode things are processed by vscode, not the backend neovim. Therefore, all neovim settings affecting insert mode don't work.
